src/protocols/device_path.rs

```diff
@@ -52,3 +52,20 @@ impl Hardware {
     pub const SUBTYPE_CONTROLLER: u8 = 0x05;
     pub const SUBTYPE_BMC: u8 = 0x06;
 }
+
+#[repr(C, packed)]
+pub struct HardDriveDevicePathNode {
+    pub header: Protocol,
+    pub partition_number: u32,
+    pub partition_start: u64,
+    pub partition_size: u64,
+    pub partition_signature: [u64; 2],
```

Comment:
Tianocore uses `[u8; 16]`, maybe we do that as well?

Reply:
Done